src/pages/groupComparison/VennSimple.tsx

```diff
@@ -0,0 +1,230 @@
+import * as React from "react";
+import {observer} from "mobx-react";
+import {computed, observable} from "mobx";
+import autobind from "autobind-decorator";
+import {ComparisonGroup} from "./GroupComparisonUtils";
+import * as d3 from 'd3';
+import _ from "lodash";
+import LazyMemo from "../../shared/lib/LazyMemo";
+
+export interface IVennSimpleProps {
+    groups: {
+        uid: string;
+        cases: string[];
+    }[]; // either 2 or 3
+    x:number;
+    y:number;
+    width:number;
+    height:number;
+    uidToGroup: { [uid: string]: ComparisonGroup };
+    onClickRegion:(regionParams:{ intersectedUids:string[], excludedUids:string[] })=>void;
+    caseCountFilterName:string;
+}
+
+function blendColors(colors:string[]) {
+    switch (colors.length) {
+        case 1:
+            return colors[0];
+        case 2:
+            return d3.interpolateLab(colors[0], colors[1])(0.5);
+        case 3:
+        default:
+            return d3.interpolateLab(
+                d3.interpolateLab(colors[0], colors[1])(0.5),
+                colors[2]
+            )(0.5);
+    }
+}
+
+function regionMouseOver(e:any) {
+    e.target.setAttribute("fill", e.target.getAttribute("data-hover-fill"));
+}
+
+function regionMouseOut(e:any) {
+    e.target.setAttribute("fill", e.target.getAttribute("data-default-fill"));
+}
+
+function getExcludedIndexes(combination:number[], numGroupsTotal:number) {
+    const excl = [];
+    for (let i=0; i<numGroupsTotal; i++) {
+        if (combination.indexOf(i) === -1) {
+            excl.push(i);
+        }
+    }
+    return excl;
+}
+
+@observer
+export default class VennSimple extends React.Component<IVennSimpleProps, {}> {
+    @observable.ref svg:SVGElement|null;
+    private regionClickCallbacks = new LazyMemo<{
+        combination:number[],
+        numGroupsTotal:number
+    }, ()=>void>(
+        params=>`${JSON.stringify(params.combination)}_${params.numGroupsTotal}`,
+        params=>{
+            return ()=>{
+                this.props.onClickRegion({
+                    intersectedUids: params.combination.map(index=>this.props.groups[index].uid),
+                    excludedUids:getExcludedIndexes(params.combination, params.numGroupsTotal).map(index=>this.props.groups[index].uid)
+                });
+            }
+        }
+    );
+
+    @autobind private svgRef(_svg:SVGElement|null) {
+        this.svg = _svg;
+    }
+
+    @computed get maxCircleRadius() {
+        return Math.min(this.props.width, this.props.height) / 4;
+    }
+
+    @computed get minCircleRadius() {
+        return this.maxCircleRadius - 30;
+    }
+
+    @computed get circleRadii() {
+        // Make a linear scale where max group is max circle radius, and max group / 10 is min circle radius
+        const radius = (numCases:number)=>{
```

Comment:
feels like it should have a test?

Reply:
I don't know how I would test this besides plugging values into the same formula. I think visual test is good enough for this. What do you think?